internal/proxy/oauthproxy.go

```diff
@@ -375,6 +375,27 @@ func (p *OAuthProxy) IsWhitelistedRequest(req *http.Request) bool {
 	return false
 }
 
+// runValidatorsWithGracePeriod runs all validators and upon finding errors, checks to see if the
+// auth provider is explicity denying authentication or if it's merely unavailable. If it's unavailable,
+// we check whether the session is within the grace period or not to determine the specific error we return.
+func (p *OAuthProxy) runValidatorsWithGracePeriod(session *sessions.SessionState) (err error) {
+	logger := log.NewLogEntry()
+	errors := options.RunValidators(p.Validators, session)
+	if len(errors) == len(p.Validators) {
```

Comment:
This logic seems wrong to me. If every validator fails, we want to return nil and allow the user?

Reply:
I think (unless _I'm_ misunderstanding/misreading) the `return nil` on line `396` is only called if _not_ all validators failed. Too many `{ }`'s everywhere 😬. Maybe a simple newline there will make that a little bit clearer.

Is that what you were looking at?

Reply:
I'm sorry -- its been a few weeks. Is the logic here OR logic, if any of the validators pass, we want to allow the user?

Reply:
No worries! That's correct - allow the user if any of the defined validators pass.